src/main/scala/edu/twitter/model/evaluation/ModelEvaluator.scala

```diff
@@ -0,0 +1,148 @@
+package edu.twitter.model.evaluation
+
+import edu.twitter.model.api.GenericModel
+import edu.twitter.model.impl.TweetsLoader
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.Row
+import org.elasticsearch.spark.rdd.EsSpark
+
+/** Representation of the records used for training
+  * and testing the model. */
+case class Record(tweetText: String, actualLabel: Double)
+
+/** Grouping of the evaluation necessary fields
+  * these fields are used for evaluating the training error. */
+case class EvaluationFields(happyCorrect: Int, happyTotal: Int, sadCorrect: Int, sadTotal: Int) {
+
+  /**
+    * Combining two evaluation instances.
+    *
+    * @param o instance to combine with this one.
+    * @return a new instance resulting of the combination of
+    *         both instances.
+    */
+  def +(o: EvaluationFields): EvaluationFields = o match {
+    case EvaluationFields(oHappyCorrect, oHappyTotal, oSadCorrect, oSadTotal) =>
+      EvaluationFields(
+        happyCorrect + oHappyCorrect,
+        happyTotal + oHappyTotal,
+        sadCorrect + oSadCorrect,
+        sadTotal + oSadTotal
+      )
+  }
+}
+
+/** Representation of a training tweet that has been classified
+  * by the model, it holds both the actual and model labels. */
+case class EvaluatedTrainingTweet(actualLabel: Double, modelPrediction: Double, tweetText: String)
+
+/**
+  * Responsible for evaluating a model based on a given
+  * testing data. this class will print training analysis
+  * in the console and persist the labeled data in `Elasticsearch`
+  * for further visualization.
+  *
+  * @param sc spark context.
+  */
+class ModelEvaluator(sc: SparkContext) {
+  /**
+    * Given a model will evaluate it based on
+    * training data and log evaluation analysis.
+    *
+    * @param model target model for evaluation.
+    */
+  def evaluate(model: GenericModel): Unit = {
```

Comment:
Sorry, what is the use of this method now?

Reply:
This method will evaluate a given `GenericModel` using data stored in `resources` we plan to use it to evaluate the models using data labeled by `textblob`

Reply:
So why tweetsLoader don't take dataPath as a parameter?

Reply:
Good point, I will add it :+1: